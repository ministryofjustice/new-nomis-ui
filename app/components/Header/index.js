import React from 'react'
import PropTypes from 'prop-types'
import { Header } from 'new-nomis-shared-components'
import { Link } from 'react-router-dom'
import ProductGlobals from '../../product-globals'
import { DesktopOnly, MobileOnly } from '../CommonTheme'
import MenuToggle from '../MenuToggle'
import MobileMenu from '../MobileMenu'
import history from '../../history'
import {
  PageHeader,
  LeftContent,
  RightContent,
  Logo,
  CrestImg,
  LogoText,
  Title,
  ToggleWrapper,
  UnstyledLink,
} from './header.theme'

const HmppsHeader = ({ user, menuOpen, setMenuOpen, extraLinks, omicUrl }) => {
  if (user && user.isKeyWorker) {
    extraLinks.push({
      text: 'My key worker allocations',
      url: `${omicUrl}key-worker/${user.staffId}`,
    })
  }

  return (
    <div>
      <DesktopOnly>
        <Header
          homeLink="/"
          title="Digital Prison Services"
          logoText="HMPPS"
          extraLinks={extraLinks}
          history={history}
          user={user}
          menuOpen={menuOpen}
          setMenuOpen={setMenuOpen}
        />
      </DesktopOnly>

      <MobileOnly>
        <PageHeader>
          <div className="header-content clickable">
            <LeftContent>
              <Link to="/">
                <Logo>
                  <CrestImg src="/img/Crest@2x.png" alt="" width="42" height="35" />
                </Logo>
              </Link>

              <UnstyledLink to="/">
                <LogoText>HMPPS</LogoText>
                <Title>{ProductGlobals.serviceName}</Title>
              </UnstyledLink>
            </LeftContent>
            <RightContent>
              <MobileOnly>
                {user && (
                  <ToggleWrapper>
                    <MenuToggle menuOpen={menuOpen} toggleMenu={() => setMenuOpen(!menuOpen)} />
                  </ToggleWrapper>
                )}
              </MobileOnly>
            </RightContent>
          </div>
          <MobileOnly>
            {menuOpen && <MobileMenu extraLinks={extraLinks} setMenuOpen={MobileMenu} user={user} />}
          </MobileOnly>
        </PageHeader>
      </MobileOnly>
    </div>
  )
}

HmppsHeader.propTypes = {
  user: PropTypes.shape({}),
  menuOpen: PropTypes.bool.isRequired,
  setMenuOpen: PropTypes.func.isRequired,
  extraLinks: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    })
  ),
  omicUrl: PropTypes.string.isRequired,
}

HmppsHeader.defaultProps = {
  user: undefined,
  extraLinks: [],
}

export default HmppsHeader
