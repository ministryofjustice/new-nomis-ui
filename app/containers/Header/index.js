import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import HeaderComponent from '../../components/Header'
import { userType } from '../../types'

const HeaderContainer = ({ user, menuOpen, setMenuOpen, prisonStaffHubUrl }) => (
  <HeaderComponent
    user={user}
    menuOpen={menuOpen}
    setMenuOpen={setMenuOpen}
    extraLinks={
      user.caseLoadOptions && user.caseLoadOptions.length > 1
        ? [{ text: 'Change caseload', url: `${prisonStaffHubUrl}change-caseload/` }]
        : []
    }
  />
)

HeaderContainer.propTypes = {
  menuOpen: PropTypes.bool,
  user: userType,
  prisonStaffHubUrl: PropTypes.string.isRequired,
}

HeaderContainer.defaultProps = {
  menuOpen: false,
  user: undefined,
}

const mapStateToProps = state => {
  const caseLoadOptions = state.getIn(['eliteApiLoader', 'User', 'CaseLoads', 'Data'])
  const user = state.getIn(['authentication', 'user'])

  return {
    menuOpen: state.getIn(['app', 'mobileMenuOpen']),
    user: {
      ...user,
      caseLoadOptions: (caseLoadOptions && caseLoadOptions.toJS()) || [],
    },
  }
}

export default connect(mapStateToProps)(HeaderContainer)
