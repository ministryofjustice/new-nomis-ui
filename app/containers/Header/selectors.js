import { createSelector } from 'reselect'

import selectUser from '../Authentication/selectors'
import { selectUserCaseLoads, selectAssignmentsTotal } from '../EliteApiLoader/selectors'

export default () =>
  createSelector(
    selectUser(),
    selectUserCaseLoads(),
    selectAssignmentsTotal(),
    (userState, caseLoadsState, rolesState, totalAssignmentsState) => {
      if (userState && caseLoadsState) {
        const allCaseLoads = caseLoadsState.toJS()
        const activeCaseLoad = allCaseLoads.filter(x => x.caseLoadId === userState.activeCaseLoadId)[0]
        const roles = rolesState ? rolesState.toJS() : []
        return {
          ...userState,
          activeCaseLoad,
          totalAssignments: totalAssignmentsState,
          caseLoads: allCaseLoads,
          caseLoadOptions: allCaseLoads.filter(x => x.caseLoadId !== userState.activeCaseLoadId),
          roles,
        }
      }
      return undefined
    }
  )
