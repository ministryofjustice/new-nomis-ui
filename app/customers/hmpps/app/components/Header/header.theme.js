import styled from 'styled-components'
import colours from 'theme/colours'
import fonts from 'theme/fonts'
import { Link } from 'react-router'
import { media } from 'utils/style-utils'
import desktop from 'theme/desktop'

export const PageHeader = styled.header`
  ${fonts.misc} display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: start;
  color: ${colours.headerTextColour};
  background-color: ${colours.headerColour};
  z-index: 999;
  height: ${desktop.headerHeight}px;
  width: 100%;

  ${media.desktop`
    justify-content: center;
  `};
`

export const LeftContent = styled.div`
  float: left;
`

export const RightContent = styled.div`
  float: right;
  height: 100%;
`

export const Logo = styled.div`
  margin-left: 10px;
  margin-right: 5px;
  float: left;
`

export const LogoText = styled(Link)`
  display: none;

  color: white;
  font-size: 26px;
  padding: 3px 15px 0 10px;
  font-weight: bold;
  border-right: solid white 1px;
  margin-right: 15px;
  text-decoration: none;

  ${media.desktop`
    display: inline-block;
  `};
`

export const Title = styled(Link)`
  text-align: left;
  padding-left: 10px;
  font-size: 26px;
  flex-grow: 1;
  text-decoration: none;

  ${media.desktop`
     padding-left: 0px;
  `};
`

export const ToggleWrapper = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  display: block;
`

export const UnstyledLink = styled(Link)`
  text-decoration: none;
  color: white;
`
