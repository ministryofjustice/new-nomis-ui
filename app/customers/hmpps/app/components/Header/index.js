import React from 'react';
import { Header } from 'new-nomis-shared-components';
import {
  DesktopOnly,
  MobileOnly,
} from 'components/CommonTheme';
import MenuToggle from 'components/MenuToggle';
import MobileMenu from 'containers/MobileMenu';
import ProductGlobals from 'product-globals';
import { Link, browserHistory } from 'react-router';
import {
  PageHeader,
  LeftContent,
  RightContent,
  Logo,
  LogoText,
  Title,
  ToggleWrapper,
  UnstyledLink,
} from './header.theme';

export default (props) => {
  const { user, menuOpen,showTermsAndConditions, setMenuOpen, navigateTo } = props;

  const extraLinks = [];

  if (user && user.isKeyWorker) {
    extraLinks.push({
      text: 'My key worker allocations',
      onClick: () => navigateTo('/myKeyWorkerAllocations'),
    });
  }

  return (<div>

    <DesktopOnly>
      <Header
        homeLink='/'
        title='Prison-NOMIS'
        logoText='HMPPS'
        extraLinks={extraLinks}
        history={browserHistory}
        {...props}
      />
    </DesktopOnly>

    <MobileOnly>
      <PageHeader>
        <div className="header-content clickable">
          <LeftContent>
            <Link to="/">
              <Logo>
                <img src="/img/Crest@2x.png" alt="" width="42" height="35" />
              </Logo>
            </Link>

            <UnstyledLink to="/">
              <LogoText>HMPPS</LogoText>
              <Title>{ProductGlobals.serviceName}</Title>
            </UnstyledLink>
          </LeftContent>
          <RightContent>
            <MobileOnly>
              { user &&
              <ToggleWrapper>
                <MenuToggle menuOpen={menuOpen} toggleMenu={() => setMenuOpen(!menuOpen)} />
              </ToggleWrapper>
              }
            </MobileOnly>
          </RightContent>
        </div>
        <MobileOnly>
          {menuOpen && <MobileMenu showTerms={showTermsAndConditions} />}
        </MobileOnly>
      </PageHeader>
    </MobileOnly>
  </div>)
}