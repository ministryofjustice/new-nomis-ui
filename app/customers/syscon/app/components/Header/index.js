import React from 'react';
import Dropdown from 'components/Dropdown';
import MenuToggle from 'components/MenuToggle';
import MobileMenu from 'containers/MobileMenu';

import './index.scss';

export default ({ user, switchCaseLoad, menuOpen, setMenuOpen }) => (
    <div className="banner">

      <div className="header-content">

           <div className="brand-header">

             <span className="section" to="/">
               <img src="/img/Syscon_logo.svg" />
             </span>

             <span className="section">
               <h1>
                 SYSCON
               </h1>
             </span>
             <span className="section desktop-only">
               <span className="divider" />
             </span>

             <span className="section desktop-only">
               <h1>
                 Prison Manager
               </h1>
             </span>
           </div>


          <div className="desktop-menu">
            {user &&
              <Dropdown
                switchCaseLoad={switchCaseLoad}
                user={user}
                menuOpen={menuOpen}
                toggleMenu={() => setMenuOpen(!menuOpen)}
              />
            }
          </div>

          <div className="mobile-menu">
            {user &&
              <MenuToggle
                menuOpen={menuOpen}
                toggleMenu={() => setMenuOpen(!menuOpen)}
              />
            }
          </div>

        <div className="mobile-only">
          {menuOpen && <MobileMenu />}
        </div>

      </div>

  </div>)
