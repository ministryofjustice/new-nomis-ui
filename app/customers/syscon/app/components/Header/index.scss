.banner {
  width: 100%;
  background: rgb(0, 39, 72);
  color: white;
  z-index: 999;

  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;

  @include media-breakpoint-down(md) {
    justify-content: start;
  }

  @include media-breakpoint-down(xs) {
    .desktop-only {
      display: none;
    }
  }

  .brand-header {
    display: flex;
    flex-direction: row;

    .section {
      margin-left: 15px;
      margin-right: 15px;

      .divider {
        border-right: solid 2px white;
        margin-top: 13px;
        height: 60%;
        display: block;
      }
    }

    h1 {
      font-size: 24px;
      font-weight: bold;
    }

    img {
      width: 40px;
      height: 40px;
      margin-top: 13px;
    }

    @include media-breakpoint-down(md) {
      h1 {
        font-size: 19px;
      }
      img {
        margin-top: 6px;
      }
    }
  }

  .desktop-menu {
    height: 60px;
    display: block;
  }

  .mobile-menu {
    display: none;
  }

  @include media-breakpoint-down(md) {
    .desktop-menu {
      display: none;
    }
    .mobile-menu {
      position: absolute;
      top: 0;
      right: 0;
      display: block;
    }
  }
}
