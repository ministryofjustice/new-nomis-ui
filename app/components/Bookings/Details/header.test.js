import React from 'react'
import { Map } from 'immutable'
import { shallow } from 'enzyme'

import Header from './header'

const allAlerts = [
  Map({ alertCode: 'HA' }),
  Map({ alertCode: 'XSA' }),
  Map({ alertCode: 'XA' }),
  Map({ alertCode: 'PEEP' }),
]
const irrelevantAlerts = [
  Map({ alertCode: 'HA', expired: true }),
  Map({ alertCode: 'XSA', expired: true }),
  Map({ alertCode: 'XR' }),
  Map({ alertCode: 'TAH' }),
]

const inmate = alerts =>
  Map({
    offenderNo: 'A1234RT',
    firstName: 'First',
    lastName: 'Last',
    facialImageId: -11,
    assignedLivingUnit: Map({ description: 'H-1-001', agencyName: 'HLI' }),
    alerts,
    keyworker: Map({ staffId: -6 }),
    activeAlertCount: 9,
    inactiveAlertCount: 8,
    iepLevel: 'Standard',
    csra: 'Medium',
    category: 'Cat D',
    getState: jest.fn(() =>
      Map({
        authentication: Map({ user: { staffId: 45 } }),
        app: { feedbackUrl: 'url' },
      })
    ),
    dispatch: jest.fn(),
    subscribe: jest.fn(),
  })

describe('Header component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <Header
        inmateData={inmate(allAlerts)}
        onImageClick={jest.fn()}
        offenderNo="A1234RT"
        onAlertFlagClick={jest.fn()}
      />
    )

    expect(wrapper.find('div.align-alerts AlertFlag')).toHaveLength(4)
    expect(wrapper).toMatchSnapshot()
  })

  it('should ignore irrelevant alert flags', () => {
    const wrapper = shallow(
      <Header
        inmateData={inmate(irrelevantAlerts)}
        onImageClick={jest.fn()}
        offenderNo="A1234RE"
        onAlertFlagClick={jest.fn()}
      />
    )

    expect(wrapper.find('div.align-alerts AlertFlag')).toHaveLength(0)
  })

  it('should render MiddleSection correctly large', () => {
    const wrapper = shallow(
      <Header
        inmateData={inmate(allAlerts)}
        onImageClick={jest.fn()}
        offenderNo="A1234RE"
        onAlertFlagClick={jest.fn()}
      />
    )

    expect(wrapper.find('div.visible-large > MiddleSection').shallow()).toMatchSnapshot()
  })

  it('should render MiddleSection correctly small', () => {
    const wrapper = shallow(
      <Header
        inmateData={inmate(allAlerts)}
        onImageClick={jest.fn()}
        offenderNo="A1234RN"
        onAlertFlagClick={jest.fn()}
      />
    )

    expect(wrapper.find('div.visible-small > MiddleSection').shallow()).toMatchSnapshot()
  })
})
