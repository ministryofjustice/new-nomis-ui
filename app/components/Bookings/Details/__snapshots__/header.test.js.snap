// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  className="header-details"
>
  <div
    className="row visible-small"
  >
    <div
      className="col-sm-12 no-padding-left"
    >
      <span
        className="acct-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        ACCT OPEN
      </span>
      <span
        className="acct-post-closure-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        ACCT POST CLOSURE
      </span>
      <span
        className="assault-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        STAFF ASSAULTER
      </span>
      <span
        className="arsonist-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <img
          alt="Arsonist"
          className="arsonist-adjust"
          src="/img/Arsonist_icon.png"
        />
        ARSONIST
      </span>
      <span
        className="disability-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <img
          alt="PEEP"
          className="disability-adjust"
          src="/img/Disability_icon.png"
        />
        PEEP
      </span>
      <span
        className="corruptor-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <img
          alt="Corruptor"
          className="corruptor-adjust"
          src="/img/CU_icon.png"
        />
        CORRUPTOR
      </span>
      <span
        className="chemical-attacker-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        CHEMICAL ATTACKER
      </span>
      <span
        className="concerted-indiscipline-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        CONCERTED INDISCIPLINE
      </span>
      <span
        className="racist-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        RACIST
      </span>
      <span
        className="risk-lgbt-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        RISK TO LGBT
      </span>
      <span
        className="hostage-taker-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        HOSTAGE TAKER
      </span>
      <span
        className="controlled-unlock-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        CONTROLLED UNLOCK
      </span>
      <span
        className="gang-member-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        GANG MEMBER
      </span>
      <span
        className="csip-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        CSIP
      </span>
      <span
        className="veteran-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        VETERAN
      </span>
      <span
        className="care-experienced-status"
        onClick={[MockFunction]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        
        CARE EXPERIENCED
      </span>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-2 col-xs-3 no-left-gutter no-right-gutter"
    >
      <div
        className="photo clickable"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="link"
        tabIndex={0}
      >
        <img
          className="sc-bdVaJa gYHwxf"
          src="/app/images/-11/data"
        />
      </div>
    </div>
    <div
      className="col-xs-9 col-sm-9 col-md-10 col-lg-10 no-left-gutter no-right-gutter add-gutter-bottom-mobile"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="inline-header-large align-alerts"
          >
            <span
              className="acct-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              ACCT OPEN
            </span>
            <span
              className="acct-post-closure-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              ACCT POST CLOSURE
            </span>
            <span
              className="assault-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              STAFF ASSAULTER
            </span>
            <span
              className="arsonist-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              <img
                alt="Arsonist"
                className="arsonist-adjust"
                src="/img/Arsonist_icon.png"
              />
              ARSONIST
            </span>
            <span
              className="disability-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              <img
                alt="PEEP"
                className="disability-adjust"
                src="/img/Disability_icon.png"
              />
              PEEP
            </span>
            <span
              className="corruptor-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              <img
                alt="Corruptor"
                className="corruptor-adjust"
                src="/img/CU_icon.png"
              />
              CORRUPTOR
            </span>
            <span
              className="chemical-attacker-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              CHEMICAL ATTACKER
            </span>
            <span
              className="concerted-indiscipline-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              CONCERTED INDISCIPLINE
            </span>
            <span
              className="racist-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              RACIST
            </span>
            <span
              className="risk-lgbt-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              RISK TO LGBT
            </span>
            <span
              className="hostage-taker-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              HOSTAGE TAKER
            </span>
            <span
              className="controlled-unlock-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              CONTROLLED UNLOCK
            </span>
            <span
              className="gang-member-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              GANG MEMBER
            </span>
            <span
              className="csip-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              CSIP
            </span>
            <span
              className="veteran-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              VETERAN
            </span>
            <span
              className="care-experienced-status"
              onClick={[MockFunction]}
              onKeyDown={[Function]}
              role="link"
              tabIndex={0}
            >
              
              CARE EXPERIENCED
            </span>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-3 col-sm-3 col-xs-12"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <span
                className="label"
              >
                Prison number
              </span>
              <strong>
                A1234RT
              </strong>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <span
                className="label"
              >
                Key worker
              </span>
              <strong>
                <span>
                  -6
                </span>
              </strong>
              <span
                className="label"
              >
                Last session
              </span>
              <strong>
                --
              </strong>
            </div>
          </div>
        </div>
        <div
          className="visible-large"
        >
          <div
            className="middle-section"
          >
            <div
              className="col-xs-4 col-sm-3 visible-large"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <span
                    className="label"
                  >
                    Incentive Level
                  </span>
                  <strong>
                    Standard
                  </strong>
                  <div>
                    <a
                      className="link"
                      data-qa="iep-details-link"
                      href="http://prisonstaffhub/offenders/A1234RT/incentive-level-details"
                    >
                      Incentive details
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <span
                    className="label"
                  >
                    CSRA
                  </span>
                  <strong>
                    Medium
                  </strong>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                  data-qa="category"
                >
                  <span
                    className="label"
                  >
                    Category
                  </span>
                  <strong>
                    Cat D
                  </strong>
                  <div>
                    <a
                      className="link"
                      data-qa="categorisation-external-link"
                      href="http://catTool100"
                    >
                      View category
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-xs-12 visible-small"
            >
              <div
                className="row"
              >
                <div
                  className="col-xs-4 d-inline-block"
                >
                  <span
                    className="label"
                  >
                    Incentive Level
                  </span>
                  <strong>
                    Standard
                  </strong>
                  <div>
                    <a
                      className="link"
                      data-qa="iep-details-link"
                      href="http://prisonstaffhub/offenders/A1234RT/incentive-level-details"
                    >
                      Incentive details
                    </a>
                  </div>
                </div>
                <div
                  className="col-xs-4 d-inline-block"
                >
                  <span
                    className="label"
                  >
                    CSRA
                  </span>
                  <strong>
                    Medium
                  </strong>
                </div>
                <div
                  className="col-xs-4 d-inline-block"
                >
                  <span
                    className="label"
                  >
                    Category
                  </span>
                  <strong>
                    Cat D
                  </strong>
                  <div>
                    <a
                      className="link"
                      data-qa="categorisation-external-link"
                      href="http://catTool100"
                    >
                      View category
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="visible-large"
            >
              <div
                className="col-xs-6 col-sm-3"
              >
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <span
                      className="label"
                    >
                      Alerts
                    </span>
                    <a
                      className="alerts-link"
                      href="/offenders/A1234RT/alerts#tab-content"
                      onClick={[Function]}
                    >
                      <span
                        className="active-alert"
                      >
                        <strong>
                          9
                        </strong>
                        <span>
                           active 
                        </span>
                      </span>
                      <span>
                        <strong
                          className="inactive-alert"
                        >
                          8
                        </strong>
                        <span>
                           inactive 
                        </span>
                      </span>
                    </a>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <div>
                      <span
                        className="label"
                      >
                        Location
                      </span>
                      <div>
                        <strong>
                          H-1-001
                        </strong>
                      </div>
                      <div>
                        <strong>
                          HLI
                        </strong>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="visible-small"
            >
              <div
                className="col-xs-4"
              >
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <span
                      className="label"
                    >
                      Alerts
                    </span>
                    <a
                      className="alerts-link"
                      href="/offenders/A1234RT/alerts#tab-content"
                      onClick={[Function]}
                    >
                      <span
                        className="active-alert"
                      >
                        <strong>
                          9
                        </strong>
                        <span>
                           active 
                        </span>
                      </span>
                      <span>
                        <strong
                          className="inactive-alert"
                        >
                          8
                        </strong>
                        <span>
                           inactive 
                        </span>
                      </span>
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="col-xs-4"
              >
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <div>
                      <span
                        className="label"
                      >
                        Location
                      </span>
                      <div>
                        <strong>
                          H-1-001
                        </strong>
                      </div>
                      <div>
                        <strong>
                          HLI
                        </strong>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-xs-12 col-sm-3"
            >
              <div
                className="stacked-links"
              >
                <div>
                  <a
                    className="button-link"
                    href="/offenders/A1234RT/add-case-note"
                    name="add-case-note-link"
                    onClick={[Function]}
                  >
                    Add case note
                  </a>
                </div>
                <div>
                  <a
                    className="button-link"
                    href="http://prisonstaffhub/offenders/A1234RT/add-appointment"
                    name="add-appointment-link"
                  >
                    Add appointment
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="visible-small"
    >
      <div
        className="middle-section"
      >
        <div
          className="col-xs-4 col-sm-3 visible-large"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <span
                className="label"
              >
                Incentive Level
              </span>
              <strong>
                Standard
              </strong>
              <div>
                <a
                  className="link"
                  data-qa="iep-details-link"
                  href="http://prisonstaffhub/offenders/A1234RT/incentive-level-details"
                >
                  Incentive details
                </a>
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <span
                className="label"
              >
                CSRA
              </span>
              <strong>
                Medium
              </strong>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col"
              data-qa="category"
            >
              <span
                className="label"
              >
                Category
              </span>
              <strong>
                Cat D
              </strong>
              <div>
                <a
                  className="link"
                  data-qa="categorisation-external-link"
                  href="http://catTool100"
                >
                  View category
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-xs-12 visible-small"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-4 d-inline-block"
            >
              <span
                className="label"
              >
                Incentive Level
              </span>
              <strong>
                Standard
              </strong>
              <div>
                <a
                  className="link"
                  data-qa="iep-details-link"
                  href="http://prisonstaffhub/offenders/A1234RT/incentive-level-details"
                >
                  Incentive details
                </a>
              </div>
            </div>
            <div
              className="col-xs-4 d-inline-block"
            >
              <span
                className="label"
              >
                CSRA
              </span>
              <strong>
                Medium
              </strong>
            </div>
            <div
              className="col-xs-4 d-inline-block"
            >
              <span
                className="label"
              >
                Category
              </span>
              <strong>
                Cat D
              </strong>
              <div>
                <a
                  className="link"
                  data-qa="categorisation-external-link"
                  href="http://catTool100"
                >
                  View category
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="visible-large"
        >
          <div
            className="col-xs-6 col-sm-3"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <span
                  className="label"
                >
                  Alerts
                </span>
                <a
                  className="alerts-link"
                  href="/offenders/A1234RT/alerts#tab-content"
                  onClick={[Function]}
                >
                  <span
                    className="active-alert"
                  >
                    <strong>
                      9
                    </strong>
                    <span>
                       active 
                    </span>
                  </span>
                  <span>
                    <strong
                      className="inactive-alert"
                    >
                      8
                    </strong>
                    <span>
                       inactive 
                    </span>
                  </span>
                </a>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div>
                  <span
                    className="label"
                  >
                    Location
                  </span>
                  <div>
                    <strong>
                      H-1-001
                    </strong>
                  </div>
                  <div>
                    <strong>
                      HLI
                    </strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="visible-small"
        >
          <div
            className="col-xs-4"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <span
                  className="label"
                >
                  Alerts
                </span>
                <a
                  className="alerts-link"
                  href="/offenders/A1234RT/alerts#tab-content"
                  onClick={[Function]}
                >
                  <span
                    className="active-alert"
                  >
                    <strong>
                      9
                    </strong>
                    <span>
                       active 
                    </span>
                  </span>
                  <span>
                    <strong
                      className="inactive-alert"
                    >
                      8
                    </strong>
                    <span>
                       inactive 
                    </span>
                  </span>
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-xs-4"
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div>
                  <span
                    className="label"
                  >
                    Location
                  </span>
                  <div>
                    <strong>
                      H-1-001
                    </strong>
                  </div>
                  <div>
                    <strong>
                      HLI
                    </strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-xs-12 col-sm-3"
        >
          <div
            className="stacked-links"
          >
            <div>
              <a
                className="button-link"
                href="/offenders/A1234RT/add-case-note"
                name="add-case-note-link"
                onClick={[Function]}
              >
                Add case note
              </a>
            </div>
            <div>
              <a
                className="button-link"
                href="http://prisonstaffhub/offenders/A1234RT/add-appointment"
                name="add-appointment-link"
              >
                Add appointment
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
